src/editor/classes/MessageHandler.js

```diff
@@ -0,0 +1,457 @@
+        //return data;
+    }
+
+    C_CMD_SET_FOCUS(data) 
+    {
+        editor.focus();
+    }
+
+    C_CMD_BLUR(data) 
+    {
+        document.activeElement.blur();
+    }
+
+    C_CMD_DISPLAY_PRINT_STYLE(data) 
+    {
+        Printer.displayPrintStyle($(".editor")[0], editor);
+    }
+
+    C_CMD_DISPLAY_NORMAL_STYLE(data) 
+    {
+        Printer.displayNormalStyle($(".editor")[0], editor);
+    }
+
+/*
+    Get the word under the (first) cursor head, or an empty string if there isn't one.
+*/
+    C_FUN_GET_CURRENT_WORD(data) 
+    {
+        var cur = editor.getCursor();
+        var line = editor.getLine(cur.line);
+
+        var strL = line.substring(0, cur.ch);
+        var strR = line.substring(cur.ch);
+        var regexL = /[^\w]?(\w*)$/;
+        var regexR = /^(\w*)[^\w]?/;
+        var matchL = regexL.exec(strL);
+        var matchR = regexR.exec(strR);
+
+        var word = "";
+        if (matchL !== null) word += matchL[1];
+        if (matchR !== null) word += matchR[1];
+
+        return word;
+    }
+
+    C_CMD_DUPLICATE_LINE(data) 
+    {
+        var cur = editor.getCursor();
+        var line = editor.getLine(cur.line);
+        var pos = {line: cur.line, ch: line.length};
+        editor.replaceRange('\n' + line, pos);
+    }
+
+    C_CMD_MOVE_LINE_UP(data) 
+    {
+        var cur = editor.getCursor();
+    
+        // check previous line is not beginning of the document
+        if ((cur.line - 1) < 0) {
+            return;
+        }
+    
+        var line = editor.getLine(cur.line) + '\n' + editor.getLine(cur.line-1);
+        var from = { line: cur.line - 1, ch: 0           };
+        var to   = { line: cur.line,     ch: line.length };
+    
+        editor.replaceRange(line, from, to);
+        editor.setCursor(cur.line - 1, cur.ch );
+    }
+
+    C_CMD_MOVE_LINE_DOWN(data) 
+    {
+        var cur = editor.getCursor();
+    
+        // check that next line is not past end of document
+        if ((cur.line + 1) == editor.lineCount() )  {
+            return;
+        }
+    
+        var line = editor.getLine(cur.line + 1) + '\n' + editor.getLine(cur.line);
+        var from = { line: cur.line,     ch: 0           };
+        var to   = { line: cur.line + 1, ch: line.length };
+    
+        editor.replaceRange(line, from, to);
+        editor.setCursor(cur.line + 1, cur.ch );
+    }
+
+
+    C_CMD_DELETE_LINE(data) 
+    {
+        editor.execCommand("deleteLine");
+        editor.changeGeneration(true);
+    }
+
+    C_CMD_TRIM_LEADING_TRAILING_SPACE(data) 
+    {
+        editLines(function (x) { return x.trim(); });
+    }
+
+    C_CMD_TRIM_TRAILING_SPACE(data) 
+    {
+        editLines(function (x) { return x.replace(/\s+$/, ""); });
+    }
+
+    C_CMD_TRIM_LEADING_SPACE(data) 
+    {
+        editLines(function (x) { return x.replace(/^\s+/, ""); });
+    }
+
+    C_CMD_TAB_TO_SPACE(data) 
+    {
+        editLines(function (x) {
+            tabToSpaceCounter = 0
```

Comment:
This comment is no longer relevant since I moved helpers to a plain functions file.  It all works as-is